packages/components/src/radio-control/types.ts

```diff
@@ -17,9 +17,9 @@ export type RadioControlProps = Pick<
 	 */
 	options?: {
 		/**
-		 * The label to be shown to the user
+		 * The label to be shown to the user.
 		 */
-		label: string;
+		label: string | React.ReactNode;
```

Comment:
`ReactNode` is a bit broad. Do we really want `label` to possibly be `boolean` or `null` or even `number`? I wonder if we meant `ReactElement` instead.

Reply:
Mmm, I don't think so. Maybe a `number,` but I believe `string` covers that.
Replacing...

Reply:
[c6b9c9e](https://github.com/WordPress/gutenberg/pull/66615/commits/c6b9c9e6f3ea978180b8805e0bc3c88638683aaa)

Reply:
> Do we really want `label` to possibly be `boolean` or `null`

This is a reasonable point, but I think we should also consider how many of our components that accept non-string labels right now already accept `ReactNode`. Introducing a subtle inconsistency like this could cause complications when composing components/types. Would it be ok to consider a package-wide switch from `ReactNode` to `ReactElement` as a separate issue?

Reply:
I'm happy to consider this a separate issue 👍 Let's document it in an issue though.